packages/common/src/downloads/helpers/canUseHubDownloadApi.ts

```diff
@@ -0,0 +1,29 @@
+import { IArcGISContext } from "../../ArcGISContext";
+import { IHubEditableContent } from "../../core/types/IHubEditableContent";
+import { isHostedFeatureServiceEntity } from "../../content/hostedServiceUtils";
+import { isMapOrFeatureServiceEntity } from "./isMapOrFeatureServiceEntity";
+
+/**
+ * @private
+ * Determines if the Hub Download API can be used for the given entity.
+ * @param entity entity to check if Hub Download API can be used
+ * @param context ArcGIS context
+ * @returns whether the Hub Download API can be used
+ */
+export function canUseHubDownloadApi(
+  entity: IHubEditableContent,
+  context: IArcGISContext
+): boolean {
+  // TODO: use permission instead.
```

Comment:
ditto

Reply:
I'm actually curious what the best path is to take here. I don't think a permission could do such complex validation on the entity types, but I guess we could create a permission that reports whether the download API is available in general. I'll take this on in a follow-on PR.